service/matching/matching_engine.go

```diff
@@ -1351,6 +1364,16 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 		}
 	}
 
+	if req.ShowTaskQueueConfig {
+		userData, _, err := pm.GetUserDataManager().GetUserData()
+		if err != nil {
+			return nil, err
+		}
+		taskQueueType := req.GetTaskQueueType()
+		taskQueueUserData := userData.GetData().GetPerType()[int32(taskQueueType)]
+		descrResp.DescResponse.Configs = taskQueueUserData.TaskQueueConfig
```

Comment:
this may be nil so you have to be careful:
```suggestion
		taskQueueType := req.GetTaskQueueType()
		taskQueueUserData := userData.GetData().GetPerType()[int32(taskQueueType)]
		descrResp.DescResponse.Configs = taskQueueUserData.GetTaskQueueConfig()
```